RELEASES.md

```diff
@@ -69,19 +69,16 @@ The release process for the main ibc-rs workspace is as follows:
       than 5 new crates or 30 existing crates by default. If we need to publish
       more than these limits, we need to update `release.toml` at workspace
       root.
-5. Run `cargo doc -p ibc --all-features --open` locally to double-check that all
```

Comment:
Removed this step as there is already a `make check-docs` rule that is specified in the `rust.yaml` file.